actor/builtin/paymentbroker/paymentbroker_test.go

```diff
@@ -446,6 +637,39 @@ func TestPaymentBrokerCloseWithCondition(t *testing.T) {
 	})
 }
 
+func TestPaymentBrokerCloseChecksCachedConditions(t *testing.T) {
+	tf.UnitTest(t)
+
+	addrGetter := address.NewForTestGetter()
+	toAddress := addrGetter()
+	method := "paramsNotZero"
+	addrParam := addrGetter()
+	sectorIdParam := uint64(6)
+	payerParams := []interface{}{addrParam, sectorIdParam}
+	blockHeightParam := types.NewBlockHeight(43)
+	redeemerParams := []interface{}{blockHeightParam}
+
+	sys := setup(t)
+	require.NoError(t, sys.st.SetActor(context.TODO(), toAddress, actor.NewActor(pbTestActorCid, types.NewZeroAttoFIL())))
+
+	// Close without params and expect a panic
+	condition := &types.Predicate{To: toAddress, Method: method}
```

Comment:
Provide params here to show that the exact same command fails before the redeem and succeeds after because the condition has been stored.